src/contour/Config.h

```diff
@@ -144,263 +189,3020 @@ struct SimpleColorConfig
+            loadFromEntry(child, "wm_class", where.wmClass);
+            loadFromEntry(child, "shell", where.shell);
+            loadFromEntry(child, "terminal_size", where.terminalSize);
+            loadFromEntry(child, "terminal_id", where.terminalId);
+            loadFromEntry(child["history"], "limit", where.maxHistoryLineCount);
+            loadFromEntry(child["history"], "scroll_multiplier", where.historyScrollMultiplier);
+            loadFromEntry(child["scrollbar"], "position", where.scrollbarPosition);
+            loadFromEntry(child["status_line"], "position", where.statusDisplayPosition);
+            loadFromEntry(child["status_line"],
+                          "sync_to_window_title",
+                          where.syncWindowTitleWithHostWritableStatusDisplay);
+            loadFromEntry(child["status_line"], "display", where.initialStatusDisplayType);
+            loadFromEntry(child["scrollbar"], "hide_in_alt_screen", where.hideScrollbarInAltScreen);
+            loadFromEntry(child, "option_as_alt", where.optionKeyAsAlt);
+            loadFromEntry(child, "history.auto_scroll_on_update", where.autoScrollOnUpdate);
+            loadFromEntry(child, "font", where.fonts);
+            loadFromEntry(child["permissions"], "capture_buffer", where.captureBuffer);
+            loadFromEntry(child["permissions"], "change_font", where.changeFont);
+            loadFromEntry(child["permissions"],
+                          "display_host_writable_statusline",
+                          where.displayHostWritableStatusLine);
+            loadFromEntry(child, "vi_mode_scrolloff", where.modalCursorScrollOff);
+            loadFromEntry(child, "slow_scrolling_time", where.smoothLineScrolling);
+            loadFromEntry(child, "vi_mode_highlight_timeout", where.highlightTimeout);
+            loadFromEntry(
+                child, "highlight_word_and_matches_on_double_click", where.highlightDoubleClickedWord);
+            loadFromEntry(child["background"], "opacity", where.backgroundOpacity);
+            loadFromEntry(child["background"], "blur", where.backgroundBlur);
+            loadFromEntry(child, "colors", where.colors);
+            loadFromEntry(child, "hyperlink_decoration.normal", where.hyperlinkDecorationNormal);
+            loadFromEntry(child, "hyperlink_decoration.hover", where.hyperlinkDecorationHover);
+            loadFromEntry(child, "bell", where.bell);
+            loadFromEntry(child, "frozen_dec_modes", where.frozenModes);
+            // YAML::Emitter emitter;
+            // emitter << child;
+            // std::cout << emitter.c_str() << std::endl;
+        }
+    }
+    void load(Config& c)
+    {
+        loadFromEntry("platform_plugin", c.platformPlugin);
+        // TODO(pr)
+        if (c.platformPlugin.get() == "auto")
+        {
+            c.platformPlugin.get() = "";
+        }
+        loadFromEntry("default_profile", c.defaultProfileName);
```

Comment:
I think it should be possible to encode the key name into the `ConfigEntry` to not just hold the entry type `T`.

`using ThatConfigEntry = ConfigEntry<That, "that">;` 

Having a look at https://ctrpeach.io/posts/cpp20-string-literal-template-parameters/ - we should probably be able to make it. :)